src/rdsn/include/dsn/utility/metrics.h

```diff
@@ -215,6 +221,54 @@ class metric_entity_prototype
     DISALLOW_COPY_AND_ASSIGN(metric_entity_prototype);
 };
 
+// `metric_timer` is a timer class that runs metric-related computations periodically, such as
```

Comment:
> runs metric-related computations periodically

Even though there isn't any thirdparty monitoring system, it will still caculate periodically? Would it be a bit of cost?
Another way is to calculate lazily, that is to say, calculate will only be triggered when any body request these metrics, you can define some callbacks for that.